src/wallets/injected/SenderWallet.ts

```diff
@@ -76,11 +74,11 @@ class SenderWallet extends InjectedWallet implements ISenderWallet {
     });
 
     if (!accessKey) {
+      if (this.wallet.isSignedIn()) await this.disconnect();
```

Comment:
I think I made a mistake reusing `InitResponse` for `requestSignIn`. I think the idea of getting no `accessKey` back is wrong for this method (only occurs for `init`). I also think disconnecting here might have some weird implications if signing out displays a prompt in the future for SenderWallet. The user would go to login only to be shown a logout flow

Reply:
If the user signs in successfully than the `InitResponse `is correct but if the user closes the window without signing in the response we get back is entirely different.
It returns only this:

```javascript
{
    "error": "User reject",
    "notificationId": 1644402625554,
    "type": "sender-wallet-result"
}
```

As for the disconnecting at this point works fine, but I don't know what the Sender Wallet has in store for the future.

Reply:
That's rather annoying. I think we need to take a closer look at the types for this wallet (if only Sender Wallet had documentation). We'll basically need to check if there's an `error` property in the response. If there is, we should throw straight away (or in this case simply return as cancelling isn't really an "error"). I'd suggest creating a separate interface for the method:

```ts
interface AccessKey =  {
  publicKey: {
    data: Uint8Array;
    keyType: number;
  };
  secretKey: string;
}

export interface InitResponse {
  accessKey: AccessKey | "";
}

export type RequestSignInResponse =
  | { accessKey: AccessKey; }
  | {
      error: string;
      notificationId: number;
      type: "sender-wallet-result";
    };
```